hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -1004,8 +1018,12 @@ private void initializeDagger(@NonNull final State state, @NonNull final InitTri
                 .round();
         final var initialStateHash = new InitialStateHash(initialStateHashFuture, roundNum);
 
-        final var activeRoster = tssBaseService.chooseRosterForNetwork(
-                state, trigger, serviceMigrator, version, configProvider.getConfiguration(), platform.getRoster());
+        final Roster activeRoster;
+        if (!configProvider.getConfiguration().getConfigData(TssConfig.class).keyCandidateRoster()) {
+            activeRoster = rosterHistory.getCurrentRoster();
+        } else {
+            activeRoster = tssBaseService.chooseRosterForNetwork(state, trigger, serviceMigrator, version);
```

Comment:
Note this TSS decision logic should actually live in the "can adopt?" `Predicate<Roster>` passed to `RosterService` (currently a no-op [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java#L458)); it will use this predicate to decide [whether to adopt the candidate roster](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/roster/schemas/V0540RosterSchema.java#L124) **during States API initialization**; so we won't need a `RosterHistory` process object at all---any component that needs the active roster can always get a correct answer from a `RosterStore`.

Reply:
Removed `RosterHistory` from `Hedera` and updated the above to get the `activeRoster` from `RosterStore`.